plugins/builds/create.js

```diff
@@ -50,9 +50,19 @@ module.exports = () => ({
                         }
                     })
                     // user has good permissions, sync and create a build
-                    .then(() => pipeline.sync())
+                    .then(() => {
+                        if (job.isPR()) {
+                            return pipeline.syncPR(job.prNum);
+                        }
+
+                        return pipeline.sync();
+                    })
                     .then(() => user.unsealToken())
-                    .then(token => scm.getCommitSha({ token, scmUri: pipeline.scmUri }))
+                    .then(token => scm.getCommitSha({
+                        token,
+                        scmUri: pipeline.scmUri,
+                        prNum: job.prNum
+                    }))
                     .then(sha => eventFactory.create({
```

Comment:
Need to pass the correct `sha` to payload so that new build create will have the correct sha.
https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/builds/create.js#L65